determine_versions.py

```diff
@@ -0,0 +1,648 @@
+            context_dict[key] = tmpl.render(context_dict)
+
+    for key in rendered_yaml:
+        render_recursive(rendered_yaml[key], context_dict, jenv)
+
+    rendered_yaml = normalize_recipe(rendered_yaml)
+    return rendered_yaml
+
+def is_new_version_available(raw_yaml, context_dict, rendered_yaml):
+    raw_yaml = copy.deepcopy(raw_yaml)
+    current_version = rendered_yaml["package"]["version"]
+    github_url = Github().get_url(rendered_yaml)
+    if github_url is not None:
+        github_version = Github().get_version(github_url)
+        if VersionOrder(github_version) > VersionOrder(current_version):
+            return True, github_version
+    else:
+        download_version = RawURL().get_url(raw_yaml, context_dict, rendered_yaml)
+        if download_version is not None:
+            if VersionOrder(download_version) > VersionOrder(current_version):
+                return True, download_version
+    return False, current_version
+
+def get_new_url_for_new_version(raw_yaml, context_dict, new_version):
+    yaml = copy.deepcopy(raw_yaml)
+    context = copy.deepcopy(context_dict)
+    yaml["context"]["version"] = new_version
+    context["version"] = new_version
+    rendered = get_rendered_yaml(yaml, context)
+    return rendered["source"][0]["url"]
+
+def get_sha256(url: str) -> Optional[str]:
+    try:
+        return hash_url(url, timeout=120, hash_type="sha256")
+    except Exception as e:
+        print("url hashing exception: %s", repr(e))
+        return None
+
+def get_updated_raw_yaml(recipe_path):
+    raw_yaml, context_dict = get_raw_yaml(recipe_path)
+
+    yaml    = copy.deepcopy(raw_yaml)
+    context = copy.deepcopy(context_dict)
+    rendered_yaml = get_rendered_yaml(yaml, context)
+
+    if "sha256" in context:
+        sha256_hash_for_current = context["sha256"]
+    else:
+        sha256_hash_for_current = rendered_yaml["source"][0]["sha256"]
+
+    version_available, new_version = is_new_version_available(yaml, context, rendered_yaml)
+    if not version_available:
+        return raw_yaml
+
+    url_for_version = get_new_url_for_new_version(yaml, context, new_version)
+    sha256_hash_for_version = get_sha256(url_for_version)
+
+    if version_available and sha256_hash_for_version is not None and sha256_hash_for_version != sha256_hash_for_current:
+        raw_yaml["context"]["version"] = new_version
+        context["version"] = new_version
+        if "sha256" in context:
+            raw_yaml["context"]["sha256"] = sha256_hash_for_version
+        else:
+            if isinstance(raw_yaml["source"], list):
+                raw_yaml["source"][0]["sha256"] = sha256_hash_for_version
+            else:
+                raw_yaml["source"]["sha256"] = sha256_hash_for_version
+        raw_yaml["build"]["number"] = 0
+    return raw_yaml
+
+if __name__ == "__main__":
+    import glob
+    loader = get_yaml_loader(typ="rt")
+    recipes = glob.glob("recipes_emscripten/**/recipe.yaml")
```

Comment:
Adapted the script from https://github.com/mamba-org/boa-forge/blob/d000999ea95a6b95d5f497e00479749f7a1b24c7/determine_versions.py#L642 to search in the `recipes_emscripten` folder.